indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/PartialDimensionDistributionTask.java

```diff
@@ -200,6 +202,10 @@ public TaskStatus runTask(TaskToolbox toolbox) throws Exception
     Preconditions.checkNotNull(partitionDimension, "partitionDimension required in partitionsSpec");
     boolean isAssumeGrouped = partitionsSpec.isAssumeGrouped();
 
+    final File tmpDir = toolbox.getIndexingTmpDir();
+    // Temporary directory is automatically removed when this IndexTask completes.
+    FileUtils.forceMkdir(tmpDir);
```

Comment:
How about pushing this code into `TaskToolbox.getIndexingTmpDir()`? We can check if the directory exists first and then create one.

Reply:
Good idea.  `forceMkdir` has guards to check if the directory already exists and is actually a directory rather than a file, so I didn't add any extra guards in TaskToolbox.